src/eos/chemTab.cpp

```diff
@@ -166,60 +201,43 @@ void ablate::eos::ChemTab::ChemTabModelComputeFunction(PetscReal density, const
     //********* Get Output tensor
     const std::size_t numOutputs = 2;
 
+    // NOTE: these names actually do make sense even though implicitly t_sourceenergy also includes inverse outputs
     TF_Output t_sourceenergy = {TF_GraphOperationByName(graph, "StatefulPartitionedCall"), 0};
     TF_Output t_sourceterms = {TF_GraphOperationByName(graph, "StatefulPartitionedCall"), 1};
 
     if (t_sourceenergy.oper == nullptr) throw std::runtime_error("ERROR: Failed TF_GraphOperationByName StatefulPartitionedCall:0");
     if (t_sourceterms.oper == nullptr) throw std::runtime_error("ERROR: Failed TF_GraphOperationByName StatefulPartitionedCall:1");
     std::array<TF_Output, numOutputs> output = {t_sourceenergy, t_sourceterms};
 
-    //********* Allocate data for inputs & outputs
+    //********* Allocate input_data for inputs & outputs
     std::array<TF_Tensor *, numInputs> inputValues = {nullptr};
     std::array<TF_Tensor *, numOutputs> outputValues = {nullptr, nullptr};
 
     std::size_t ndims = 2;
 
     // according to Varun this should work for including Zmix
     auto ninputs = progressVariablesNames.size();
-    int64_t dims[] = {1, (int)ninputs};
-    float data[ninputs];
+    int64_t dims[] = {(int)batch_size, (int)ninputs};
+    float input_data[batch_size][ninputs];  // NOTE: we changed the 1st dimension from 1 --> batch_size
 
-    for (std::size_t i = 0; i < ninputs; i++) {
-        data[i] = (float)(densityProgressVariable[i] / density);
-    }
+    // NOTE: this should be modified sufficiently to support batch inputs!
+    for (size_t i = 0; i < batch_size; i++)
+        for (std::size_t j = 0; j < ninputs; j++) input_data[i][j] = (float)(densityProgressVariables[i][j] / density[i]);
 
-    std::size_t ndata = ninputs * sizeof(float);
-    TF_Tensor *input_tensor = TF_NewTensor(TF_FLOAT, dims, (int)ndims, data, ndata, &NoOpDeallocator, nullptr);
+    std::size_t ndata = sizeof(input_data);
+    TF_Tensor *input_tensor = TF_NewTensor(TF_FLOAT, dims, (int)ndims, input_data, ndata, &NoOpDeallocator, nullptr);
```

Comment:
I think we might be able to save time by reusing the memory/tensor object?  We could do that by having the ChemTabSourceCalculator own the tensor.  The ChemTabSourceCalculator instance is always the same batch size.